src/xenia/config.h

```diff
@@ -0,0 +1,42 @@
+#ifndef XENIA_CONFIG_H_
+#define XENIA_CONFIG_H_
+#include <cpptoml/include/cpptoml.h>
+#include <string>
+
+namespace xe {
+
+// The main type that runs the whole emulator.
+// This is responsible for initializing and managing all the various subsystems.
+class Config {
+ public:
+  static Config& GetInstance() { return (*instance_); }
+  static void Init(const std::wstring& content_path);
+  static void Shutdown();
+  void SaveConfig();
+  void RestoreDefaults();
+
+  // config variables
+  std::string gpu_;
```

Comment:
Not really sure about having things like this public, usually getters are created for this. Still, why a class rather than a namespace?